examples/go-client/README.md

```diff
@@ -20,18 +17,61 @@ Auth API clients must perform two-way authentication using x509 certificates:
 2. They must offer their x509 certificate, which has been previously issued
    by the auth sever.
 
-Start up a teleport auth server and then run the following commands to create an `api-user` with a signed certificate. Make sure to run this from the `go-client` directory for proper output.
+The client supports multiple credential loading methods to authenticate the client. Multiple methods can be used simultaneously in order to increase redundancy or to handle special cases where some authentication methods are expected to fail/expire sometimes.
```

Comment:
would an average user provision multiple distinct sets of credentials?
if it were me, I'd create one type of credentials and make sure they stay valid. it's much less work than provisioning multiple distinct ones.

Reply:
I agree and I use it the same way, but it's a requirement to enable multiple sets of credentials. I can remove it from this example if it's too complicated, but the fact that `Credentials` is a list could become confusing if not addressed here.

Reply:
Maybe this README is just missing a section explaining the different auth methods and when a user would use one or the other.
Then, in that section you could explain that they can be combined (and when to do that)